### PR TITLE
Add a per-SteamClient unique identifier message to DebugLog messages (2)

### PR DESCRIPTION
**Pull request by yaakov-h**
A different approach to solving #503.

Thoughts?

This version should not trigger breaking changes in consumers.

cc/ @JustArchi @xPaw

### CODE REVIEW

File: SteamKit2/SteamKit2/Steam/CMClient.cs [hunk 1]
```diff
@@ -23,13 +23,18 @@ namespace SteamKit2.Internal
...
     {
         /// <summary>
         /// The configuration for this client.
         /// </summary>
         public SteamConfiguration Configuration { get; }
 
+        /// <summary>
+        /// A unique identifier for this client instance.
+        /// </summary>
+        public string ID { get; }
```
Comment (xPaw) [inline SteamKit2/SteamKit2/Steam/CMClient.cs]: Allow setter?

Comment (JustArchi) [inline SteamKit2/SteamKit2/Steam/CMClient.cs]: All fine but a setter would indeed be nice, see my reasoning in https://github.com/SteamRE/SteamKit/issues/503#issuecomment-549124002

Comment (yaakov-h) [reply SteamKit2/SteamKit2/Steam/CMClient.cs]: Should be immutable for the object’s lifetime, so perhaps a constructor parameter?

Comment (yaakov-h) [reply SteamKit2/SteamKit2/Steam/CMClient.cs]: See above.

Comment (xPaw) [reply SteamKit2/SteamKit2/Steam/CMClient.cs]: That works.

**Pull request merged**